pwiz_tools/Skyline/Executables/SkylineBatch/SkylineBatch/Program.cs

```diff
@@ -0,0 +1,202 @@
+﻿/*
+ * Original author: Ali Marsh <alimarsh .at. uw.edu>,
+ *                  MacCoss Lab, Department of Genome Sciences, UW
+ * Copyright 2020 University of Washington - Seattle, WA
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+using System;
+using System.Configuration;
+using System.Deployment.Application;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading;
+using System.Windows.Forms;
+using log4net;
+using log4net.Appender;
+using log4net.Config;
+using log4net.Repository.Hierarchy;
+using SkylineBatch.Properties;
+
+namespace SkylineBatch
+{
+    class Program
+    {
+        private static readonly ILog Log = LogManager.GetLogger("SkylineBatch");
+        private static string _version;
+        
+
+        [STAThread]
+        public static void Main(string[] args)
+        {
+            Application.EnableVisualStyles();
+            Application.SetCompatibleTextRenderingDefault(false);
+            Application.SetUnhandledExceptionMode(UnhandledExceptionMode.CatchException);
+            // Handle exceptions on the UI thread.
+            Application.ThreadException += ((sender, e) => Log.Error(e.Exception));
+            // Handle exceptions on the non-UI thread.
+            AppDomain.CurrentDomain.UnhandledException += ((sender, e) =>
+            {
+                try
+                {
+                    Log.Error(Resources.SkylineBatch_encountered_an_unexpected_error, (Exception)e.ExceptionObject);
+                    MessageBox.Show(Resources.SkylineBatch_encountered_an_unexpected_error +
+                                    @"Error details may be found in the SkylineBatchProgram.log file in this directory : "
+                                    + Path.GetDirectoryName(Application.ExecutablePath)
+                    );
+                }
+                finally
+                {
+                    Application.Exit();
+                }
+            });
+
+            using (var mutex = new Mutex(false, $"University of Washington {AppName()}"))
+            {
+                if (!mutex.WaitOne(TimeSpan.Zero))
+                {
+                    MessageBox.Show($@"Another instance of {AppName()} is already running.", $@"{AppName()} Error", MessageBoxButtons.OK, MessageBoxIcon.Error);

```

Comment:
I don't think you should prevent more than one instance of SkylineBatch running at the same time.
Does anything bad happen if I run more than one instance of SkylineBatch at the same time? Are they all saving their stuff to the same place?